client/my-sites/domains/paths.js

```diff
@@ -121,10 +121,29 @@ export function domainMapping( siteName, domain = '' ) {
 	return path;
 }
 
-export function domainTransferIn( siteName, domain ) {
+/**
+ * Return the path to start an inbound domain transfer to WordPress.com.
+ *
+ * @param { string } siteName         The slug for the site.
+ * @param { string } domain           The domain name.
+ * @param { boolean } useStandardBack Flag to indicate whether the "Back" button in the
+ *                                      transfer page should return to the current URL context.
+ * @returns { string } Path to the inbound domain transfer UI.
+ */
+export function domainTransferIn( siteName, domain, useStandardBack ) {
 	let path = `/domains/add/transfer/${ siteName }`;
+	const params = [];
+
 	if ( domain ) {
-		path += `?initialQuery=${ domain }`;
+		params.push( `initialQuery=${ domain }` );
+	}
+
+	if ( useStandardBack ) {
+		params.push( 'useStandardBack=1' );
+	}
+
+	if ( params.length > 0 ) {
+		path += '?' + params.join( '&' );
```

Comment:
At this point, I'd recommend using [`qs`](https://github.com/ljharb/qs) - we already use it in Calypso, so it's not a new dependency.

Example usage would be:

```
const query = {};
if ( domain ) {
    params.initialQuery = domain;
}

if ( useStandardBack ) {
    query.useStandardBack = true; // or 1
}

const queryString = qs.stringify( query );
if ( queryString ) {
    path += '?' + queryString;
}
```

Not exactly much shorter, but definitely less hacky and we get some security benefits as well, I'd imagine.